### DISCUSSION TITLE
DiaNN formats conversion

### DISCUSSION THREAD

**Discussion by Arthfael** [vdemichev/DiaNN#612] [General]
I would like to start the discussion on DIA-NN's output files and their compatibility with other proteomics tools or public repositories - my apologies if it exists somewhere already, I could not find it.
Briefly: DiaNN is fantastic, we are relying on it increasingly at our facility. But there are two limitations that make it limiting: both the reports and library format are very unique.
- I understand that the speclib format is very "lightweight/efficient". However it is not easy to browse or convert into other formats. 
- Whilst the reports are very useful, they currently do not allow a full-submission on PRIDE (which would require mzIdentML or mzTab files).
If there are tools available to convert the DiaNN formats to more broadly used equivalents, I will be happy to stand corrected. Otherwise, what would be the best solution for addressing these limitations of an otherwise fantastic piece of software? Is anyone out there working on conversion tools from DiaNN formats to more prevalent Proteomics formats?

**Comment by vdemichev** [comment]
- DIA-NN itself can convert .speclib to .tsv, so if you'd like to submit a large (e.g. full predicted proteome) library to PRIDE, can indeed do this in .speclib format.
- Yes, I am not sure if anyone actually is doing full submissions of DIA data to PRIDE? I would check the quantms project https://nf-co.re/quantms, I think they might have a DIA-NN-based pipeline for this.

**Reply by PavelBouchal** [reply]
Hi Vadim, you wrote "DIA-NN itself can convert .speclib to .tsv" Can you specify exactly how this can be done?
Many thanks
Pavel

**Reply by vdemichev** [reply]
Hi Pavel, just need to specify the .speclib in the Spectral Library field, and select 'Generate spectral library'. The Raw files field must be empty.

**Reply by PavelBouchal** [reply]
Yes, this works! Many thanks.

**Comment by Arthfael** [comment]
Dear Vadim,
Wow, that was fast. Thank you very much for the quick response and sorry
for the trouble.
Also, thanks for this awesome software!
Kind regards,

Armel

**Comment by Arthfael** [comment]
Dear Vadim,

And what about the reverse conversion, i.e. .tsv (or other) to .speclib? Is
there a way to do it? Tsv is convenient for merging/filtering databases,
but of course makes for horrendously large files, any way to "compress" to
speclib would be fantastic.
Kind regards,

Armel

**Reply by vdemichev** [reply]
Hi Armel,

.tsv to .speclib happens automatically once the .tsv is loaded into DIA-NN.

Best,
Vadim

**Comment by Arthfael** [comment]
Oh, so same as speclib to tsv, do a run without raw files?

**Comment by Arthfael** [comment]
Actually, no, I have tried that recently and it did not work. But as long
as we complete 1 search with 1 file with the tsv library, we will get our
speclib too.
Thanks!
